console crashes when unlocking non-existant account (Görli)

#### System information

Geth version: 1.9.15-unstable-3666da8a-20200514
OS & Version: x86_64 GNU/Linux 5.4

#### Expected behaviour

It should give a recoverable error, something like "no accounts present, please create an account"

#### Actual behaviour

Runtime error:
````

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0xe90ce3]

goroutine 89 [running]:
github.com/dop251/goja.(*Runtime).RunProgram.func1(0xc000371d60)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/runtime.go:889 +0x98
panic(0x1069f60, 0x1c71510)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/dop251/goja.(*vm).try.func1(0xc000297040, 0x0, 0xc000371c30, 0x0, 0x0, 0x0, 0xc000371ca8)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/vm.go:377 +0x3da
panic(0x1069f60, 0x1c71510)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/ethereum/go-ethereum/console.(*bridge).UnlockAccount(0xc000334ff0, 0x1494cc0, 0xc0007747a0, 0xc0008fe020, 0x1, 0x1c, 0xc000702780, 0x1201170, 0x1, 0x1f28520, ...)
        /home/eth/go-ethereum/console/bridge.go:236 +0xa3
github.com/ethereum/go-ethereum/internal/jsre.MakeCallback.func1(0x1494cc0, 0xc0007747a0, 0xc0008fe020, 0x1, 0x1c, 0x14953a0, 0x1f28520)
        /home/eth/go-ethereum/internal/jsre/jsre.go:264 +0x9f
github.com/dop251/goja.(*vm)._nativeCall(0xc000297040, 0xc000689760, 0x1)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/vm.go:1839 +0x291
github.com/dop251/goja.call.exec(0x1, 0xc000297040)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/vm.go:1823 +0x4b5
github.com/dop251/goja.(*vm).run(0xc000297040)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/vm.go:289 +0x99
github.com/dop251/goja.(*vm).try(0xc000297040, 0xc000371cb0, 0x0)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/vm.go:383 +0x132
github.com/dop251/goja.(*vm).runTry(0xc000297040, 0x0)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/vm.go:388 +0x4e
github.com/dop251/goja.(*Runtime).RunProgram(0xc000702780, 0xc00017e0c0, 0x0, 0x0, 0x0, 0x0)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/runtime.go:900 +0x1c9
github.com/dop251/goja.(*Runtime).RunScript(0xc000702780, 0x0, 0x0, 0xc00003c0f0, 0x28, 0xc000000180, 0x3, 0x0, 0x0)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/runtime.go:879 +0x9d
github.com/dop251/goja.(*Runtime).RunString(...)
        /home/eth/go/pkg/mod/github.com/dop251/goja@v0.0.0-20200219165308-d1232e640a87/runtime.go:868
github.com/ethereum/go-ethereum/internal/jsre.(*JSRE).Evaluate.func1(0xc000702780)
        /home/eth/go-ethereum/internal/jsre/jsre.go:275 +0x5c
github.com/ethereum/go-ethereum/internal/jsre.(*JSRE).runEventLoop(0xc0003b5d40)
        /home/eth/go-ethereum/internal/jsre/jsre.go:204 +0x601
created by github.com/ethereum/go-ethereum/internal/jsre.New
        /home/eth/go-ethereum/internal/jsre/jsre.go:79 +0x14b
````

#### Steps to reproduce the behaviour

Start geth with no accounts configured

`$ geth --goerli`

attach a console

`$ geth attach $HOME/.ethereum/goerli/geth.ipc`

type in the console:

`personal.unlockAccount(eth.accounts[0])`

crash happens

#### Backtrace

````
above
````

--- a/console/bridge.go
+++ b/console/bridge.go
@@ -233,11 +233,12 @@ func (b *bridge) UnlockAccount(call jsre.Call) (goja.Value, error) {
 	if len(call.Arguments) < 1 {
 		return nil, fmt.Errorf("usage: unlockAccount(account, [ password, duration ])")
 	}
+
+	account := call.Argument(0)
 	// Make sure we have an account specified to unlock.
-	if call.Argument(0).ExportType().Kind() != reflect.String {
+	if goja.IsUndefined(account) || goja.IsNull(account) || account.ExportType().Kind() != reflect.String {
 		return nil, fmt.Errorf("first argument must be the account to unlock")
 	}
-	account := call.Argument(0)
 
 	// If password is not given or is the null value, prompt the user for it.
 	var passwd goja.Value
@@ -285,10 +286,10 @@ func (b *bridge) Sign(call jsre.Call) (goja.Value, error) {
 		passwd  = call.Argument(2)
 	)
 
-	if message.ExportType().Kind() != reflect.String {
+	if goja.IsUndefined(message) || message.ExportType().Kind() != reflect.String {
 		return nil, fmt.Errorf("first argument must be the message to sign")
 	}
-	if account.ExportType().Kind() != reflect.String {
+	if goja.IsUndefined(account) || account.ExportType().Kind() != reflect.String {
 		return nil, fmt.Errorf("second argument must be the account to sign with")
 	}
 
@@ -317,10 +318,11 @@ func (b *bridge) Sleep(call jsre.Call) (goja.Value, error) {
 	if nArgs := len(call.Arguments); nArgs < 1 {
 		return nil, fmt.Errorf("usage: sleep(<number of seconds>)")
 	}
-	if !isNumber(call.Argument(0)) {
+	sleepObj := call.Argument(0)
+	if goja.IsUndefined(sleepObj) || goja.IsNull(sleepObj) || !isNumber(sleepObj) {
 		return nil, fmt.Errorf("usage: sleep(<number of seconds>)")
 	}
-	sleep := call.Argument(0).ToFloat()
+	sleep := sleepObj.ToFloat()
 	time.Sleep(time.Duration(sleep * float64(time.Second)))
 	return call.VM.ToValue(true), nil
 }
@@ -338,13 +340,13 @@ func (b *bridge) SleepBlocks(call jsre.Call) (goja.Value, error) {
 		return nil, fmt.Errorf("usage: sleepBlocks(<n blocks>[, max sleep in seconds])")
 	}
 	if nArgs >= 1 {
-		if !isNumber(call.Argument(0)) {
+		if goja.IsNull(call.Argument(0)) || goja.IsUndefined(call.Argument(0)) || !isNumber(call.Argument(0)) {
 			return nil, fmt.Errorf("expected number as first argument")
 		}
 		blocks = call.Argument(0).ToInteger()
 	}
 	if nArgs >= 2 {
-		if !isNumber(call.Argument(1)) {
+		if goja.IsNull(call.Argument(1)) || goja.IsUndefined(call.Argument(1)) || !isNumber(call.Argument(1)) {
 			return nil, fmt.Errorf("expected number as second argument")
 		}
 		sleep = call.Argument(1).ToInteger()

--- a/console/bridge_test.go
+++ b/console/bridge_test.go
@@ -1,0 +1,48 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package console
+
+import (
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/ethereum/go-ethereum/internal/jsre"
+)
+
+// TestUndefinedAsParam ensures that personal functions can receive
+// `undefined` as a parameter.
+func TestUndefinedAsParam(t *testing.T) {
+	b := bridge{}
+	call := jsre.Call{}
+	call.Arguments = []goja.Value{goja.Undefined()}
+
+	b.UnlockAccount(call)
+	b.Sign(call)
+	b.Sleep(call)
+}
+
+// TestNullAsParam ensures that personal functions can receive
+// `null` as a parameter.
+func TestNullAsParam(t *testing.T) {
+	b := bridge{}
+	call := jsre.Call{}
+	call.Arguments = []goja.Value{goja.Null()}
+
+	b.UnlockAccount(call)
+	b.Sign(call)
+	b.Sleep(call)
+}

